command/agent/agent.go

```diff
@@ -309,6 +309,8 @@ func convertServerConfig(agentConfig *Config) (*nomad.Config, error) {
 	conf.RPCAddr.IP = rpcAddr.IP
 	conf.SerfConfig.MemberlistConfig.BindPort = serfAddr.Port
 	conf.SerfConfig.MemberlistConfig.BindAddr = serfAddr.IP.String()
+	conf.SerfConfig.RejoinAfterLeave = agentConfig.Server.RejoinAfterLeave
+	// todo: surely there are other SerfConfig fields missing and this probably shouldn't be so heavily drilled-through
```

Comment:
this is more a note to self that should be squashed out of the commit

Reply:
I think we're ok here -- most of the fields don't have Nomad-specific overrides, so we're starting from the default config value.